pkg/controller/master/node/promote_controller.go

```diff
@@ -288,6 +290,8 @@ func selectPromoteNode(nodeList []*corev1.Node) *corev1.Node {
 			managementNumber++
 		}
 
+		witnessPromoted = witnessPromoted || isWitnessNode(node, isManagement)
```

Comment:
if `witnessPromoted` is true, then it's always true, is that expected?

Reply:
yes, this means this round, the selection will know we already have one witness node promoted.